include/nbl/video/utilities/CComputeBlit.h

```diff
@@ -0,0 +1,635 @@
+			return EF_R16G16_SNORM;
+
+		default:
+			return EF_UNKNOWN;
+		}
+	}
+
+	static inline const char* getGLSLFormatStringFromFormat(const asset::E_FORMAT format)
+	{
+		const char* result;
+		switch (format)
+		{
+		case asset::EF_R32G32B32A32_SFLOAT:
+			result = "rgba32f";
+			break;
+		case asset::EF_R16G16B16A16_SFLOAT:
+			result = "rgba16f";
+			break;
+		case asset::EF_R32G32_SFLOAT:
+			result = "rg32f";
+			break;
+		case asset::EF_R16G16_SFLOAT:
+			result = "rg16f";
+			break;
+		case asset::EF_B10G11R11_UFLOAT_PACK32:
+			result = "r11f_g11f_b10f";
+			break;
+		case asset::EF_R32_SFLOAT:
+			result = "r32f";
+			break;
+		case asset::EF_R16_SFLOAT:
+			result = "r16f";
+			break;
+		case asset::EF_R16G16B16A16_UNORM:
+			result = "rgba16";
+			break;
+		case asset::EF_A2B10G10R10_UNORM_PACK32:
+			result = "rgb10_a2";
+			break;
+		case asset::EF_R8G8B8A8_UNORM:
+			result = "rgba8";
+			break;
+		case asset::EF_R16G16_UNORM:
+			result = "rg16";
+			break;
+		case asset::EF_R8G8_UNORM:
+			result = "rg8";
+			break;
+		case asset::EF_R16_UNORM:
+			result = "r16";
+			break;
+		case asset::EF_R8_UNORM:
+			result = "r8";
+			break;
+		case asset::EF_R16G16B16A16_SNORM:
+			result = "rgba16_snorm";
+			break;
+		case asset::EF_R8G8B8A8_SNORM:
+			result = "rgba8_snorm";
+			break;
+		case asset::EF_R16G16_SNORM:
+			result = "rg16_snorm";
+			break;
+		case asset::EF_R8G8_SNORM:
+			result = "rg8_snorm";
+			break;
+		case asset::EF_R16_SNORM:
+			result = "r16_snorm";
+			break;
+		case asset::EF_R8_UINT:
+			result = "r8ui";
+			break;
+		case asset::EF_R16_UINT:
+			result = "r16ui";
+			break;
+		case asset::EF_R32_UINT:
+			result = "r32ui";
+			break;
+		case asset::EF_R32G32_UINT:
+			result = "rg32ui";
+			break;
+		case asset::EF_R32G32B32A32_UINT:
+			result = "rgba32ui";
+			break;
+		default:
+			__debugbreak();
+		}
+
+		return result;
+	}
+
+private:
+	enum E_BLIT_TYPE : uint8_t
+	{
+		EBT_REGULAR = 0,
+		EBT_COVERAGE_ADJUSTMENT,
+		EBT_COUNT
+	};
+
+	core::smart_refctd_ptr<video::IGPUDescriptorSetLayout> m_blitDSLayout[EBT_COUNT];
+	core::smart_refctd_ptr<video::IGPUDescriptorSetLayout> m_kernelWeightsDSLayout;
+
+	core::smart_refctd_ptr<video::IGPUPipelineLayout> m_blitPipelineLayout[EBT_COUNT];
+	core::smart_refctd_ptr<video::IGPUPipelineLayout> m_coverageAdjustmentPipelineLayout;
+
+	core::smart_refctd_ptr<video::IGPUBuffer> m_dummySSBO = nullptr;
+
+	const uint32_t sharedMemorySize;
+	core::smart_refctd_ptr<video::ILogicalDevice> device;
+
+	core::smart_refctd_ptr<video::IGPUSampler> sampler = nullptr;
+
+	// Todo(achal): Isn't this just getRealWindowSize?
+	static inline core::vectorSIMDu32 getWindowDim(const core::vectorSIMDu32& inImageExtent, const core::vectorSIMDu32& outImageExtent)
+	{
+		core::vectorSIMDf scale = static_cast<core::vectorSIMDf>(inImageExtent).preciseDivision(static_cast<core::vectorSIMDf>(outImageExtent));
+		return static_cast<core::vectorSIMDu32>(core::ceil(scale));
+	}
```

Comment:
I think we don't really need to pass scaled kernels to `CBlitUtilities::getRealWindowSize` [here](https://github.com/Devsh-Graphics-Programming/Nabla/blob/037a3d07fad5ca2653017fef56152de3b78e5c40/include/nbl/asset/filters/CBlitImageFilter.h#L161). If we can get the window dimensions like I do here, I can easily use that function here and delete this one?

Reply:
no, because not all kernels have a support from `-0.5,0.5`

Also this means your entire code is wrong because it wont work with anything else than a Box filter.

Reply:
Resolved.